### PR TITLE
Fix `check-stylish-haskell.nix`

### PR DESCRIPTION
**Pull request by coot**
This fixes a mistake in #3698 which matched all the files when it should only match `ouroboros-network` subdirectory.  This is because `-p ouroboros-network` matches full path, and the parent directory is called `ouroboros-network`.

### CODE REVIEW

**Review by dnadales — APPROVED** [review]

**Comment by coot** [comment]
I still have a problem with this nix script.  When I run:
```
fd -p ouroboros-network-framework -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell-network.yaml
```
Then I get a massive diff, but `nix-build -A checks.styles.check-stylish-network` succeeds.

**Review by amesgen — CHANGES_REQUESTED** [review]
> I still have a problem with this nix script. When I run:
> 
> ```
> fd -p ouroboros-network-framework -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell-network.yaml
> ```
> 
> Then I get a massive diff, but `nix-build -A checks.styles.check-stylish-network` succeeds.

I think the problem here is that `echo $? > $out` is not actually checking anything, as `stylish-haskell -i` will succeed with exit code 0 even if it reformatted files. To fix this, we could use `git diff --exit-code` instead (but `git` has to be added to the derivation for this).

**Comment by coot** [comment]
@amesgen afaik that's indeed the problem, I reported this on [slack](https://input-output-rnd.slack.com/archives/CDA6LUXAQ/p1650527906313109?thread_ts=1648694931.837859&cid=CDA6LUXAQ).

File: nix/check-stylish.nix [hunk 1]
```diff
@@ -8,5 +8,15 @@ runCommand "check-stylish" {
...
+  fd -p ouroboros-consensus-byron -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-byronspec -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-cardano -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-cardano-test -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-mock -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-mock-test -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-protocol -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-shelley -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-shelley-test -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
+  fd -p ouroboros-consensus-test -e hs -E Setup.hs -X stylish-haskell -c .stylish-haskell.yaml -i
```
[the remark this answers is not in the document]
Comment (coot) [reply nix/check-stylish.nix:20]: You're right, it already matches all the files.

[the remark this answers is not in the document]
Comment (coot) [reply nix/check-stylish.nix]: Fixed.

**Comment by coot** [comment]
I changed the title and description to match the current state of the patch.  We still need make sure the script fails when it returns.

**Review by amesgen — APPROVED** [review]

**Comment by amesgen** [comment]
Mostly superseded by #3722

**Pull request closed without merging**